pixeltable/type_system.py

```diff
@@ -166,22 +166,14 @@ def __str__(self) -> str:
         return self._type.name.lower()
 
     def __eq__(self, other: object) -> bool:
-        return self.matches(other) and self.nullable == other.nullable
+        return isinstance(other, ColumnType) and self.matches(other) and self.nullable == other.nullable
 
-    def is_supertype_of(self, other: ColumnType) -> bool:
-        if type(self) != type(other):
-            return False
-        if self.matches(other):
-            return True
-        return self._is_supertype_of(other)
-
-    @abc.abstractmethod
-    def _is_supertype_of(self, other: ColumnType) -> bool:
-        return False
+    def is_supertype_of(self, other: ColumnType, ignore_nullable: bool = False) -> bool:
```

Comment:
We should reevaluate whether we want this flag at all. From the perspective of function calls, whether something is null or not is really outside of the type hierarchy (if a function has a non-nullable parameter and the argument in that row is null, the call is skipped anyway).